components/mpas-seaice/driver/ice_comp_mct.F

```diff
@@ -1970,13 +1972,25 @@ subroutine ice_import_mct(x2i_i, errorCode)!{{{
 
         if (trim(config_ocean_surface_type) == "free") then ! free surface (MPAS-O)
         
-           meltingPotential            = x2i_i % rAttr(index_x2i_Fioo_q, n)
+           ! freezingMeltingPotential(i) is the ocean energy associated with frazil formation 
+           ! when it is positive and frazilMassFlux is positive, and negative  when there is 
+           ! a melting potential in which case frazilMassFlux is zero
 
-           ! frazil calculation
-           frazilProduction            = x2i_i % rAttr(index_x2i_Fioo_frazil, n)
-           call freezing_potential(freezingPotential, frazilProduction, seaSurfaceSalinity(i), &
+           freezingMeltingPotential(i) = x2i_i % rAttr(index_x2i_Fioo_q, n)
+
+           frazilMassFlux              = x2i_i % rAttr(index_x2i_Fioo_frazil, n)
+
+           ! We calculate a second energy associated with frazil production that is 
+           ! the sea ice model's version, which is passed to sea ice column physics as 
+           ! the frazilProduction, which is positive when frazil forms, otherwise zero.
+           ! frazilProduction is used to balance the mass budget according to what 
+           ! the sea ice model needs to complete the conversion of frazil to sea ice for
+           ! the given quantity of energy the ocean model has passed to the sea ice model.
+
+           call frazil_mass(freezingMeltingPotential(i), frazilMassFluxRev, seaSurfaceSalinity(i), &
                                    config_thermodynamics_type)
-           freezingMeltingPotential(i) = freezingPotential + min(meltingPotential,0.0_RKIND)
+
+           frazilMassAdjust(i) = frazilMassFlux-frazilMassFluxRev 
```

Comment:
Could you explain this in more detail?

Reply:
Please see the revised documentation.  I set this up as a draft, and it was not ready for review at the time you looked at this.  However, I believe the in-code documentation answers your questions.